packages/gatsby-plugin-sitemap/src/internals.js

```diff
@@ -29,6 +29,10 @@ export const runQuery = (handler, query, excludes, pathPrefix) =>
       return page
     })
 
+    if (!r.data.site.siteMetadata.siteUrl) {
+      throw new Error(`SiteMetaData 'siteUrl' property is required`)
+    }
+
     if (r.data.site.siteMetadata.siteUrl) {
```

Comment:
I guess we could remove this `if` around `r.data.site.siteMetadata.siteUrl` for removing trailing slashes as well?   Happy to do it if you would like?

Reply:
it makes sense if we do throw earlier